app/src/common/hooks/useFhirResourceTreeData.ts

```diff
@@ -81,34 +87,65 @@ const isElementArray = ({ max, sliceName }: IElementDefinition): boolean =>
 
 const createElementNode = (
   elementDefinition: IElementDefinition,
-  isArrayItem?: boolean
+  params: { index?: number; parentPath?: string }
 ): ElementNode => {
+  const { index, parentPath } = params;
+  const prefixPath =
+    (parentPath || elementDefinition.path?.split(".").shift()) ?? "";
+  const suffixPath = elementDefinition.path?.split(".").slice(1).join(".");
+  const elementPath = `${prefixPath}${
+    prefixPath && suffixPath && "."
+  }${suffixPath}${index !== undefined ? `[${index}]` : ""}`;
   return {
     id: uuid(),
     name: elementDefinition.id?.split(".").pop() ?? "",
     children: [],
-    path: elementDefinition.path ?? "",
+    path: elementPath,
     definition: elementDefinition,
-    isArray:
-      undefined !== isArrayItem
-        ? !isArrayItem
-        : isElementArray(elementDefinition),
+    isArray: index !== undefined ? false : isElementArray(elementDefinition),
     isSlice: !!elementDefinition.sliceName,
     kind: getKind(elementDefinition),
     type: elementDefinition.type?.map((t) => computeType(t)).join(" | "),
   };
 };
 
-const getChildrenChoices = (
+const getNewPath = (path: string): string => {
+  if (path.endsWith(`]`)) {
+    const splittedPath = path.split(".");
+    const lastPathElement = splittedPath.pop()?.split("[")[0];
+    if (lastPathElement) splittedPath.push(lastPathElement);
+    return splittedPath.join(".");
+  } else {
+    return path;
+  }
+};
+
+const createElementDefinition = (attribute: Attribute): IElementDefinition => {
+  const elementDefinition: IElementDefinition = {
+    path: attribute.path,
+    id: getNewPath(attribute.path),
+    type: [{ code: attribute.definition_id }],
+  };
+  if (attribute.slice_name) elementDefinition.sliceName = attribute.slice_name;
```

Comment:
You could do directly set `elementDefinition.sliceName` like this :

```suggestion
    type: [{ code: attribute.definition_id }],
    sliceName: attribute.slice_name
  };
```

Reply:
If I do this, the elementDefinition in the elementNode.definition will have an empty string for sliceName

Reply:
Are you sure? `attribute.slice_name` can be either string or undefined when `elementDefinition.sliceName` has the same type. I don't see where is the problem..

Reply:
Yes i'm sure :) but we can also work with that as the string is empty 
![image](https://user-images.githubusercontent.com/47388675/119646143-1d505680-be1f-11eb-9a35-915e8618f6f1.png)